### PR TITLE
Abstract ranklib

### PR DESCRIPTION
**Pull request by binarymax**
This cleans up the KCV/Ranklib/Train methods into one consistent method for use with all notebooks.  Also added some TQDM progress bars.

### CODE REVIEW

**Review by nathancday — APPROVED** [review]
Looks great, thanks for exposing the kcv arg and saving off the models. And I see that 'brad pitt' query you slipped in there...

File: notebooks/solr/tmdb/tale-of-two-queries (Solr).ipynb [hunk 1]
```diff
@@ -214,7 +216,7 @@
    "outputs": [],
    "source": [
     "from ltr.search import search\n",
-    "search(client, keywords=\"drama\", modelName=\"genre\")"
+    "search(client, keywords=\"blah blahbity blah\", modelName=\"genre\")"
```
Comment (worleydl) [inline notebooks/solr/tmdb/tale-of-two-queries (Solr).ipynb:219]: Any reason for changing this?

File: notebooks/solr/tmdb/tale-of-two-queries (Solr).ipynb [hunk 2]
```diff
@@ -299,7 +316,7 @@
    "outputs": [],
    "source": [
     "from ltr.search import search\n",
-    "search(client, keywords=\"drama\", modelName=\"genre\")"
+    "search(client, keywords=\"blah\", modelName=\"genre\")"
```
Comment (worleydl) [inline notebooks/solr/tmdb/tale-of-two-queries (Solr).ipynb]: Another blah

File: ltr/client/elastic_client.py [hunk 3]
```diff
@@ -8,6 +8,8 @@
 import json
 from elasticsearch import Elasticsearch
 
+from tqdm import tqdm
```
Comment (worleydl) [inline ltr/client/elastic_client.py]: Ever used?

**Review by worleydl — COMMENTED** [review]
Added some comments/questions.

File: notebooks/solr/tmdb/tale-of-two-queries (Solr).ipynb (hunk 1 above)
Comment (nathancday) [reply notebooks/solr/tmdb/tale-of-two-queries (Solr).ipynb:219]: this calls out the lack of keyword feature in the model

**Review by nathancday — APPROVED** [review]
Tests are green, fingers crossed.

**Pull request merged**